### PR TITLE
don't post event if the stage deploys no code

### PR DESCRIPTION
**Pull request by irwaters**

### CODE REVIEW

**Comment by grosser** [comment]
test ? :trollface:

File: plugins/samson_ledger/lib/samson_ledger/client.rb [hunk 1]
```diff
@@ -19,7 +19,8 @@ def plugin_enabled?
 
       def post_deployment(deploy)
         return false unless plugin_enabled?
-        post_event(deploy)
+        # only post the event if we are deploying code
+        post_event(deploy) unless deploy.stage.no_code_deployed?
```
Comment (grosser) [inline plugins/samson_ledger/lib/samson_ledger/client.rb]: comment kind of only repeats the code ... so meh ...

File: plugins/samson_ledger/test/lib/samson_ledger/client_test.rb [hunk 2]
```diff
@@ -59,6 +63,15 @@
       SamsonLedger::Client.post_deployment(deploy)
       assert_requested(@event_sent)
     end
+    it "posts an event with a valid client" do
```
Comment (grosser) [inline plugins/samson_ledger/test/lib/samson_ledger/client_test.rb]: delete this ... just copy-paste from above ?

File: plugins/samson_ledger/test/lib/samson_ledger/client_test.rb [hunk 3]
```diff
@@ -59,6 +63,15 @@
...
+      SamsonLedger::Client.post_deployment(deploy)
+      assert_requested(@event_sent)
+    end
+
+    it "does not post an event when no_code_deployed" do
+      SamsonLedger::Client.post_deployment(no_code_deploy)
+      assert_not_requested(@event_sent)
```
Comment (grosser) [inline plugins/samson_ledger/test/lib/samson_ledger/client_test.rb]: is there a `refute_requested` ?

**Comment by grosser** [comment]
👍

File: plugins/samson_ledger/test/lib/samson_ledger/client_test.rb (hunk 3 above)
Comment (irwaters) [inline plugins/samson_ledger/test/lib/samson_ledger/client_test.rb]: nope, you get `assert_requested` and `assert_not_requested`

**Pull request merged**